src/Controls/src/Core/BindableLayout/BindableLayout.cs

```diff
@@ -490,6 +493,17 @@ View CreateEmptyView(object emptyView, DataTemplate dataTemplate)
 		}
 
 		void ItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+		{
+			if (_dispatcher?.IsDispatchRequired ?? false)
+			{
+				_dispatcher.Dispatch(() => UpdateCollection(e));
+				return;
+			}
+
+			UpdateCollection(e);
+		}
+
+		void UpdateCollection(NotifyCollectionChangedEventArgs e)
 		{
 			if (!_layoutWeakReference.TryGetTarget(out IBindableLayout layout))
```

Comment:
For example, use the `(layout as BindableObject)?.Dispatcher` here instead and move the logic into a new method. This way the `BindableLayoutController` does not need to get access to more things that are not needed directly.